### PR TITLE
TSD-139 add new scope variable

### PR DESCRIPTION
**Pull request by esoliani**
@uniphil I think the problem that the watch of selectedCode is not working when changing the year, is because the value of the variable does not actually change, so I added a new variable which has the code and the year. this way it works and fix the issue describe in the jira.
Let me know if you want to keep it like that o change it.

### CODE REVIEW

File: app/scripts/controllers/dashboard.coffee [hunk 1]
```diff
@@ -326,8 +326,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
           loadingSrv.containerLoad schoolsP, document.getElementById mapId
           schoolsP.then (schools) ->
             if $scope.selected? and $scope.viewMode == 'schools'
+              $scope.selectedCodeYear =
```
Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee]: Still looking at this commit to understand what the problem was, but,
We should try to only write to variables on `$scope` with a `watchCompute`

File: app/scripts/controllers/dashboard.coffee [hunk 2]
```diff
@@ -326,8 +326,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
...
+                code: $scope.selected.CODE
+                year: $scope.year
               $scope.select $scope.selected.CODE
 
+        $scope.$watch 'selectedCodeYear', (selectedCodeYear) -> if selectedCodeYear?
+          (findSchool selectedCodeYear.code).then (school)->
+            $scope.selected = school
```
Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee:318]: 'selected' already has a `watchCompute`, we should avoid writing to it anywhere else

**Comment by uniphil** [comment]
@esoliani check out the [`'selected'` watchCompute](https://github.com/WorldBank-Transport/edudash/blob/edudash-2.0/app/scripts/controllers/dashboard.coffee#L284-L293)

``` cofffeescript
         watchCompute 'selected',
           dependencies: ['selectedCode', 'viewMode']
```

You're right, it won't update when the year updates, because its only dependencies are `selectedCode` and `viewMode`, which will stay the same.

The problem is actually `findSchool` (and also `findPoly`) -- it uses `$scope.allSchools`, so actually `selected` depends on that.

So I think the right fix is to add `allSchools` to `dependencies`. It might be a good idea to also fix `findSchool` to take `(code, schools)` as params so it can be a pure function, but that would require more code changes.

File: app/scripts/controllers/dashboard.coffee (hunk 2 above)
Comment (esoliani) [inline app/scripts/controllers/dashboard.coffee:318]: removed

**Comment by esoliani** [comment]
changed as proposed

File: app/scripts/controllers/dashboard.coffee (hunk 2 above)
Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee:318]: it's still in the PR... maybe it only worked because this was still in the code? :(

Comment (esoliani) [inline app/scripts/controllers/dashboard.coffee:318]: yes, that is was the reason It works when we talk. my bad. We need to figure out how to implement it.

Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee:318]: damn. ok. I'll try to see if I can find out why.

**Comment by uniphil** [comment]
After looking at it for a bit today, this problem runs fairly deep. I think once some other things are cleaned up and fixed (like `findSchool` being a pure function), solving this will be more approachable.

I added some TODO comments about it. It works now, so I'm going to merge.

**Pull request merged**